src/main/java/newitem/NewItemView.java

```diff
@@ -0,0 +1,84 @@
+package newitem;
+
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.Observable;
+import java.util.Observer;
+
+
+public class NewItemView extends JFrame implements Observer, ActionListener {
+
+    private final NewItemController controller;
+
+    private JTextField nameField;
+
+    private JTextField upcField;
+
+    private JTextField priceField;
+
+    public NewItemView(NewItemController controller) {
+        this.controller = controller;
+        init();
+    }
+
+    @Override
+    public void update(Observable o, Object arg) {
+        NewItemViewModel viewModel = (NewItemViewModel) o;
+
+        setVisible(viewModel.isVisible());
+
+        if (viewModel.getStatusChanged()) {
+            if (viewModel.getStatus() == NewItemStatus.REPEAT_UPC) {
+                JOptionPane.showMessageDialog(this, "UPC already exists", "Error", JOptionPane.ERROR_MESSAGE);
+            }
+            if (viewModel.getStatus() == NewItemStatus.SUCCESS) {
+                JOptionPane.showInternalMessageDialog(this, "New product has been successfully added", "Success", JOptionPane.INFORMATION_MESSAGE);
```

Comment:
change showInternalMessageDialog to showMessageDialog, it throws an error right now, then everything is good